src/tests/roc_audio/test_depacketizer.cpp

```diff
@@ -122,6 +124,37 @@ void expect_flags(Depacketizer& depacketizer,
     }
 }
 
+class StatusReader : public packet::IReader {
+public:
+    StatusReader(packet::IReader& reader, status::StatusCode code)
+        : reader_(reader)
+        , code_(code)
+        , enabled_(false) {
+    }
+
+    virtual status::StatusCode read(packet::PacketPtr& pp) {
+        if (enabled_) {
+            return code_;
+        }
+
+        return reader_.read(pp);
+    }
+
+    void enable() {
+        enabled_ = true;
+    }
+
+    void disable() {
+        enabled_ = false;
+    }
```

Comment:
enable() and disable() looks quite unclear in test code.

Thoughts:

* mask / unmask
* enable_errors / disable_errors

Also, consider moving status code from ctor to enable() method, it may improve readability:

```cpp
reader.enable_errors(status::StatusUnknown);
```